### PR TITLE
Unset authconfig fields on disable

### PR DESCRIPTION
**Pull request by mrajashree**

### CODE REVIEW

**Comment by alena1108** [comment]
@mrajashree is it possible to add a test?

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/issues/20323

File: pkg/auth/providers/common/actions.go [hunk 1]
```diff
@@ -20,7 +20,13 @@ func HandleCommonAction(actionName string, action *types.Action, request *types.
...
 		if e, ok := config[client.AuthConfigFieldEnabled].(bool); ok && e {
 			config[client.AuthConfigFieldEnabled] = false
-			logrus.Infof("Disabling auth provider %v", authConfigName)
+			retainFields := map[string]bool{"apiVersion": true, "kind": true, "metadata": true, "type": true}
+			for field := range config {
+				if !retainFields[field] {
+					delete(config, field)
+				}
+			}
+			logrus.Infof("Disabling auth provider now%v", authConfigName)
```
Comment (alena1108) [inline pkg/auth/providers/common/actions.go]: we should remove "now" from the log statement

**Comment by mrajashree** [comment]
Need to add a validation test for this, as we need to pass auth provider details for setting up auth, which validation tests already do by accepting them through env vars. Will put a separate PR for the test

**Pull request merged**